### PR TITLE
ceph: 15.2.8 -> 15.2.10 and python dependency fix

### PR DESCRIPTION
**Pull request by srhb**
###### Motivation for this change
- Normal update, fixes at least the mentioned CVE though.
- Drop redundant patch.
- There was a regression in master that caused the mgr to fail achieving healthy state with restful because of a change in how deps are propagated. This should fix it such that tests pass, though the way the python deps are handled is still not great. One Day:tm:

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/ceph/default.nix [hunk 1]
```diff
@@ -134,7 +135,6 @@ in rec {
 
     patches = [
       ./0000-fix-SPDK-build-env.patch
-      ./ceph-glibc-2-32-sigdescr_np.patch
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/ceph/default.nix:137]: Is that patch unused? If so it should be deleted.

Comment (srhb) [reply pkgs/tools/filesystems/ceph/default.nix:137]: Ah yes, brainfart, thanks :)

**Comment by risicle** [comment]
@srhb presuming the CVE you're referring to is https://nvd.nist.gov/vuln/detail/CVE-2020-25678 ?

**Comment by risicle** [comment]
Full rebuild works for me, linux x86_64, but the ceph nixos tests have never worked for me... :(

**Pull request merged**